foundry/gui/MainWindow.py

```diff
@@ -139,7 +140,7 @@ def __init__(self, path_to_rom=""):
         """
         file_menu.addSeparator()
         settings_action = file_menu.addAction("&Settings")
-        settings_action.triggered.connect(show_settings)
+        settings_action.triggered.connect(lambda: show_settings(self))
```

Comment:
What is the difference here? Isn't `self` always implicitly passed through?

Reply:
> What is the difference here? Isn't `self` always implicitly passed through?

No. `self` is implicitly passed in when calling an object's method. `show_settings` is just a method defined in `SettingsDialog.py`, and it just creates a new `QDialog` and launches it via `.exec_()`. If it is not given the parent window object, the new `QDialog` does not inherit the stylesheet from the `MainWindow`.

Reply:
Does that mean that the code did not work before this commit?

Reply:
The code worked to create the settings dialog, but if you were using the dark theme, the settings dialog was created with the default light theme stylesheet.

By changing it to pass in the `MainWindow` object via `self`, we can now call `setStyleSheet` that changes all the window themes when toggling the radio buttons in the settings dialog.